### PR TITLE
Pnc visit location module

### LINKED ISSUE #7: Implement Capturing the location of the visit
This will allow the CHW to capture the GPS of the location where the visit is taking place

### PR DESCRIPTION
**Pull request by maulidkategile**
**PR INFO**
- Closes #7

### CODE REVIEW

File: opensrp-chw/src/nacp/assets/json.form-sw/pnc_hv_location.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        "openmrs_entity": "concept",
+        "openmrs_entity_id": "",
+        "type": "spacer",
+        "spacer_height": "15dp"
+      },
+      {
+        "key": "gps",
+        "openmrs_entity_parent": "",
+        "openmrs_entity": "concept",
+        "openmrs_entity_id": "163277AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
```
Comment (cozej4) [inline opensrp-chw/src/nacp/assets/json.form-sw/pnc_hv_location.json]: We are phasing out the use of openmrs uuids, please update this to the same value as key

File: opensrp-chw/src/nacp/assets/json.form/pnc_hv_location.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        "openmrs_entity": "concept",
+        "openmrs_entity_id": "",
+        "type": "spacer",
+        "spacer_height": "15dp"
+      },
+      {
+        "key": "gps",
+        "openmrs_entity_parent": "",
+        "openmrs_entity": "concept",
+        "openmrs_entity_id": "163277AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
```
Comment (cozej4) [inline opensrp-chw/src/nacp/assets/json.form/pnc_hv_location.json]: We are phasing out the use of openmrs uuids, please update this to the same value as key

File: opensrp-chw/src/nacp/java/org/smartregister/chw/util/NacpCoreConstants.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+public class NacpCoreConstants extends CoreConstants {
+    public static class PncHomeVisit{
+        private static final String LOCATION = "pnc_hv_location";
+
+        public static String getLocation() {
+            return Utils.getLocalForm(LOCATION, locale, assetManager);
+        }
+    }
+}
```
Comment (cozej4) [inline opensrp-chw/src/nacp/java/org/smartregister/chw/util/NacpCoreConstants.java]: Move this implementation to [Constants.java](opensrp-chw/src/main/java/org/smartregister/chw/util/Constants.java)

File: opensrp-chw/src/nacp/java/org/smartregister/chw/interactor/PncHomeVisitInteractorFlv.java [hunk 4]
```diff
@@ -136,6 +138,48 @@ private void refreshActionList() {
...
+                }
+            }
+
+            @Override
+            public BaseAncHomeVisitAction.Status evaluateStatusOnPayload() {
+                if (StringUtils.isBlank(gpsLocation))
+                    return BaseAncHomeVisitAction.Status.PENDING;
+                return BaseAncHomeVisitAction.Status.COMPLETED;
+            }
+        };
```
Comment (cozej4) [inline opensrp-chw/src/nacp/java/org/smartregister/chw/interactor/PncHomeVisitInteractorFlv.java]: I would suggest to move this implementation to a class in `actionhelper` package so as to make this file more readable.

@gosso22 What are your thoughts on this?

**Review by cozej4 — CHANGES_REQUESTED** [review]

**Comment by maulidkategile** [comment]
@cozej4 Feedback Implemented

**Review by issyzac — DISMISSED** [review]
LGTM

File: opensrp-chw/src/nacp/java/org/smartregister/chw/interactor/PncHomeVisitInteractorFlv.java (hunk 4 above)
Comment (gosso22) [reply opensrp-chw/src/nacp/java/org/smartregister/chw/interactor/PncHomeVisitInteractorFlv.java]: @cozej4 - I with this

**Comment by gosso22** [comment]
This looks good to me. @cozej4 - when you have time, can you look at the changes @maulidkategile made on this PR and approve if they are good.

**Review by cozej4 — DISMISSED** [review]

**Review by gosso22 — APPROVED** [review]

**Pull request merged**